File: pset1/cash/cash.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>

/*Coins:
    Quarter 25
    Dime 10
    Nickel 5
    Penny 1
*/

int main(void)
{
    int coin = 0;

    float change;
    do
    {
        change = get_float("Change: ");
    }
    while (change < 0);

    //Change to input to all cents
    change = roundf(change * 100);
    //Use the largest coin possible
    //How many coins to be returnwd
    //as well as remains of amounts

    while (change >= 25)
    {
        coin++;
        change -= 25;
    }

    while (change >= 10)
    {
        coin++;
        change -= 10;
    }

    while (change >= 5)
    {
        coin++;
        change -= 5;
    }

    while (change >= 1)
    {
        coin++;
        change -= 1;
    }

    //Print the final numbers of coins
    printf("%i\n", coin);
}

File: pset6/cash/cash.py
from cs50 import get_float


def main():
    # Get the user change
    change = 0
    coin = 0

    while True:
        change = get_float("Change: ")
        change = round(change * 100)

        if change > 0:
            break

    while change > 0:

        # Wokr out the change
        if change >= 25:
            coin += 1
            change -= 25
        elif change >= 10:
            coin += 1
            change -= 10
        elif change >= 5:
            coin += 1
            change -= 5
        elif change >= 1:
            coin += 1
            change -= 1

        # Print the result
        if change <= 0:
            break

    print(coin)


if __name__ == '__main__':
    main()